# HG changeset patch
# User Michael Creel  <michael.creel@uab.es>
# Date 1301405791 21600
#      Tue Mar 29 07:36:31 2011 -0600
# Node ID 17997376291bcf74af3ca1159ded8023ddd0d332
# Parent  39ca02387a320fade9682082c841eb2cdb6a3eb7
Fix variable name clash in ols.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2011-03-29  Michael Creel <michael.creel@uab.es>
+
+	* statistics/base/ols.m: Fix erroneous degrees of freedom when
+	computing the covariance estimator.
+
 2011-03-28  Rik  <octave@nomad.inbox5.com>
 
 	* linear-algebra/cross.m, plot/ishold.m, signal/fftfilt.m: Improve
 	docstrings.
 
 2011-03-26  Robert T. Short  <octave@phaselockedsystems.com.com>
 
 	* signal/ifftshift.m: Add tests.
diff --git a/scripts/statistics/base/ols.m b/scripts/statistics/base/ols.m
--- a/scripts/statistics/base/ols.m
+++ b/scripts/statistics/base/ols.m
@@ -96,29 +96,29 @@ function [beta, sigma, r] = ols (y, x)
 
   [nr, nc] = size (x);
   [ry, cy] = size (y);
   if (nr != ry)
     error ("ols: number of rows of X and Y must be equal");
   endif
 
   z = x' * x;
-  r = rank (z);
+  rnk = rank (z);
 
-  if (r == nc)
+  if (rnk == nc)
     beta = inv (z) * x' * y;
   else
     beta = pinv (x) * y;
   endif
 
   if (isargout (2) || isargout (3))
     r = y - x * beta;
   endif
   if (isargout (2))
-    sigma = r' * r / (nr - r);
+    sigma = r' * r / (nr - rnk);
   endif
 
 endfunction
 
 %!test
 %! x = [1:5]';
 %! y = 3*x + 2;
 %! x = [x, ones(5,1)];
